packages/flame/lib/src/game/notifying_vector2.dart

```diff
@@ -0,0 +1,190 @@
+import 'dart:typed_data';
+import 'package:flutter/foundation.dart';
+import 'package:vector_math/vector_math_64.dart';
+
+/// Extension of the standard [Vector2] class, implementing the [ChangeNotifier]
+/// functionality. This allows any interested party to be notified when the
+/// value of this vector changes.
+///
+/// This class can be used as a regular [Vector2] class. However, if you do
+/// subscribe to notifications, don't forget to eventually unsubscribe in
+/// order to avoid resource leaks.
+///
+/// Direct access to this vector's [storage] is not allowed.
+///
+// ignore: prefer_mixin
```

Comment:
we should consider disable this rule haha

Reply:
will do